routing/payment_session.go

```diff
@@ -12,12 +10,71 @@ import (
 // to prevent an HTLC being failed if some blocks are mined while it's in-flight.
 const BlockPadding uint16 = 3
 
-var (
+// noRouteError encodes a non-critical error encountered during path finding.
+type noRouteError uint8
+
+const (
+	// errNoTlvPayload is returned when the destination hop does not support
+	// a tlv payload.
+	errNoTlvPayload noRouteError = iota
```

Comment:
Why move these constants out of `pathfind.go`? That is where they are generated

Reply:
I think it makes sense here, since they are part of the interface definition.

Reply:
I see them as basic outputs of pathfinding, independent of how pathfinding is used.